### PR TITLE
stop running tasks when deleting a task definition if force is passed

### PR DESCRIPTION
**Pull request by fishnix**

### CODE REVIEW

File: orchestration/orchestration_taskdef.go [hunk 1]
```diff
@@ -233,21 +249,81 @@ func (o *Orchestrator) DeleteTaskDef(ctx context.Context, input *TaskDefDeleteIn
...
+					Reason:  aws.String(reason),
+					Task:    aws.String(splitTask[1]),
+				}); err != nil {
+					log.Errorf("failed calling StopTask for %s", t)
+					return
+				}
+			}
+
+			// wait for tasks to become STOPPED
+			if err := retry(10, 10*time.Second, func() error {
```
Comment (btassone) [inline orchestration/orchestration_taskdef.go:279]: Code smell. What's the upper limit on aws for attempts to stop a task?

Comment (btassone) [reply orchestration/orchestration_taskdef.go:279]: (maximum time, not retries, sorry)

File: orchestration/orchestrator.go [hunk 2]
```diff
@@ -104,3 +105,30 @@ func defaultRbfunc(name string) rollbackFunc {
 		return nil
 	}
 }
+
+type stop struct {
+	error
+}
+
+// retry is stolen from https://upgear.io/blog/simple-golang-retry-function/
```
Comment (btassone) [inline orchestration/orchestrator.go:113]: :trollface:

**Review by btassone — APPROVED** [review]
Approved. Looks good, just have a question.

File: orchestration/orchestration_taskdef.go (hunk 1 above)
Comment (fishnix) [reply orchestration/orchestration_taskdef.go:279]: I'm not sure I understand what you're getting at, this retries the function 10 times with a sliding window.

Comment (btassone) [reply orchestration/orchestration_taskdef.go:279]: Ahh nevermind, ignore that comment.

File: orchestration/orchestration_taskdef.go [hunk 3]
```diff
@@ -233,21 +249,81 @@ func (o *Orchestrator) DeleteTaskDef(ctx context.Context, input *TaskDefDeleteIn
 		}(taskDefinitionRevisions[1:])
 	}
 
-	deletedCluster, err := o.deleteCluster(ctx, &input.Cluster)
-	if err != nil {
-		return nil, fmt.Errorf("failed to delete cluster: %s", err)
-	}
+	// stop the running tasks and cleanup in the background
+	go func() {
+
```
Comment (tenyo) [inline orchestration/orchestration_taskdef.go:254]: ✂️

**Review by tenyo — APPROVED** [review]
Looks good 👍 
If this background thread fails for some reason to stop the running tasks, there's no way currently to know that from the UI, right? The task def will just remain there. I guess eventually all these async tasks should be using flywheel

**Comment by fishnix** [comment]
@tenyo yes correct - i didnt want to introduce flywheel to this API though just as a one off

**Pull request merged**